lib/runCode.js

```diff
@@ -173,14 +173,17 @@ module.exports = function (opts, cb) {
       // create a callback for async opFunc
       if (opInfo.async) {
         args.push(function (err, result) {
+          if (err) return cb(err)
+
           // save result to the stack
           if (result) {
             // NOTE: Ensure that every stack item is padded to 256 bits.
             //       This should be done at every opcode in the future.
             result = utils.setLengthLeft(result, 32)
             runState.stack.push(result)
           }
-          cb(err)
+
+          cb()
```

Comment:
Good catch, though that file has all the callbacks without null, need to tackle it at some point. Also some other methods (`parseVmResults`) there have an issue passing both `err` and `results`. Haven't reviewed it yet.